src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -3,27 +3,79 @@
 #include "geom/point.h"
 #include "geom/polygon.h"
 #include "geom/util.h"
+#include "shared/constants.h"
 
 class Obstacle
 {
    public:
+    static Obstacle createPlaceholderObstacle();
+
+    static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
+
     /*
      * Gets the boundary polygon around the given robot obstacle that other robots
-     * should not enter with a buffer scaled by radiusScaling and
-     * velocity_projection_scaling
+     * should not enter with a buffer scaled by radius_scaling and
+     * velocity_cushion_scaling
      *
      * @param robot robot to create obstacle boundary polygon around
-     * @param radiusScaling safety factor to vary the radius size of the buffer zone
-     * (1=default), must be greater than 0
-     * @param velocity_projection_scaling scales the projection buffer in the direction of
-     * the velocity (1=default), must be greater than 0
+     * @param radius_scaling multiplicatively scales the radius of the obstacle
+     * @param velocity_cushion_scaling multiplicatively scales the cushion of
+     * velocity of the robot
+     * @param enable_velocity_cushion if true, enables the cushion of the velocity
+     * of the robot as component of the obstacle
+     *
+     * @note the radius is by default the minimum distance needed for two robots to clear
+     * each other
+     * @note the velocity cushion is by default the distance that can be travelled in
+     * one tick
      *
      * @return a six-sided Polygon to represent the boundary around the obstacle
      */
-    static Polygon getBoundaryPolygon(const Robot& robot, double robot_radius_scaling,
-                                      double velocity_projection_scaling);
+    static Obstacle createRobotObstacleWithScalingParams(
+        const Robot& robot, bool enable_velocity_cushion,
+        double radius_cushion_scaling = 1.0, double tick_length = 1.0,
+        double velocity_cushion_scaling = 1.0);
+
+    /*
+     * Gets the boundary polygon around the given robot obstacle that other robots
+     * should not enter with a buffer scaled by radius_scaling and
+     * velocity_cushion_scaling
+     *
+     * @param robot robot to create obstacle boundary polygon around
+     * @param radius_buffer additively increases the radius of the obstacle by specified
+     * amount
+     * @param velocity_cushion_buffer additively increases the cushion of velocity
+     * of the robot
+     * @param enable_velocity_cushion if true, enables the cushion of the velocity
+     * of the robot as component of the obstacle
+     *
+     * @note the radius is by default the minimum distance needed for two robots to clear
+     * each other
+     * @note the velocity cushion is by default the distance that can be travelled in
+     * one tick
+     *
+     * @return a six-sided Polygon to represent the boundary around the obstacle
+     */
+    static Obstacle createRobotObstacleWithBufferParams(
```

Comment:
It might be cleaner to remove the default values and force the user to specify a radius_cushion_buffer and velocity_cushion_buffer. This makes the function more intentional and removed the extra bool parameter

Reply:
OK done